"[" should be escaped in sed expression. This patch is suitable for opkg-0.2.2

diff -Nur utils.orig/update-alternatives utils/update-alternatives
--- utils.orig/update-alternatives	2013-08-16 04:22:29.000000000 +0800
+++ utils/update-alternatives	2014-09-19 10:55:22.238159317 +0800
@@ -68,6 +68,10 @@
 	sed -e 's/\//\\\//g'
 }
 
+protect_special_character() {
+	sed -e 's/\[/\\\[/g'
+}
+
 remove_alt() {
 	[ $# -lt 2 ] && return 1
 	local name="$1"
@@ -75,7 +79,7 @@
 	
 	[ ! -f $ad/$name ] && return 0
 
-	path=`echo $path | protect_slashes`
+	path=`echo $path | protect_slashes | protect_special_character`
 	sed -ne "/^$path\>.*/!p" $ad/$name > $ad/$name.new
 	mv $ad/$name.new $ad/$name
 }
